plugins/idaskins/clrapplier.py

```diff
@@ -0,0 +1,124 @@
+
+__all__ = ['load_clr_file']
+
+
+def _ida_lib():
+    ea_name = 'ida64' if idc.__EA64__ else 'ida'
+    if sys.platform == 'win32':
+        functype = WINFUNCTYPE
+        lib = WinDLL(ea_name)
+    elif sys.platform == 'darwin':
+        functype = CFUNCTYPE
+        lib = CDLL(idaapi.idadir("lib" + ea_name + ".dylib"))
+    else:
+        functype = CFUNCTYPE
+        lib = CDLL('lib' + ea_name + '.so')
+    return functype, lib
+
+
+functype, lib = _ida_lib()
+hook_cb_t = functype(c_void_p, c_void_p, c_int, c_void_p)
+
+hook_to_notification_point = lib.hook_to_notification_point
+hook_to_notification_point.argtypes = [c_int, hook_cb_t, c_void_p]
+
+unhook_from_notification_point = lib.unhook_from_notification_point
+unhook_from_notification_point.argtypes = [c_int, hook_cb_t, c_void_p]
+
+
+class TemporaryFilter(QObject):
+    """
+    Temporary event filter installed at qApp to catch events
+    while executing QDialog::exec.
+
+    The filter automatically clicks &Import button,
+    and automatically selects file by using native ui hooks.
+    """
+
+    def __init__(self, filepath):
+        super(TemporaryFilter, self).__init__()
+        filepath = os.path.abspath(filepath)
+        assert os.path.isfile(filepath)
+
+        self.filepath = filepath
+
+    def eventFilter(self, obj, event):
+        def is_colors_dialog():
+            return isinstance(
+                obj, QDialog) and 'IDA Colors' in obj.windowTitle()
+
+        if isinstance(event, QResizeEvent) and is_colors_dialog():
+            obj.setWindowFlags(Qt.Window | Qt.FramelessWindowHint)
+            obj.setFixedSize(0, 0)
+            obj.setGeometry(0, 0, 0, 0)
+            event.accept()
+            return 1
+        if isinstance(event, QFocusEvent) and is_colors_dialog():
+            qApp.removeEventFilter(self)
+            buttons = [widget for widget in obj.children() if isinstance(
+                widget, QDialogButtonBox)][0]
+            button = buttons.buttons()[-4]
+
+            with NativeHook(ask_file=self.ask_file_handler):
+                button.click()
+
+            QTimer.singleShot(0, lambda: obj.accept())
+            return 1
+        return 0
+
+    def ask_file_handler(self):
+        return create_string_buffer(self.filepath)
+
+
+class NativeHook:
+    """
+    Installer for non-exposed hooks from UI_Hooks.
+    This uses hook_to_notification_point with HT_UI.
+
+    with NativeHook(ask_file=lambda: 0):
+        # do anything
+    """
+    NAMES = {
+        'ask_file': 0x1d
+    }
+    HT_UI = 1
+
+    def __init__(self, **kwargs):
+        self.hooks = {NativeHook.NAMES[key]: value for key, value in kwargs.items()}
+        self._handler = hook_cb_t(self.handler)
+
+    def handler(self, _user_data, code, _va_args):
+        if code in self.hooks:
+            try:
+                res = self.hooks[code]()
+                return cast(res, c_void_p).value
+            except:
```

Comment:
Can this except statement be a bit more specific, e.g. just catch `KeyError`?

Reply:
In this line, I tried to return NULL pointer if any exception has occurred. I just tried `raise Exception()` outside try-except and IDA stuck in an infinite loop which is weird behavior...

Reply:
Alright, I see! IDA's Py embedding is not always perfect, printing a stack-trace is probably the best we can do here.